downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/MapOldStableIds.java

```diff
@@ -0,0 +1,123 @@
+import java.nio.file.StandardCopyOption;
+import java.nio.file.StandardOpenOption;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+
+public class MapOldStableIds {
+	private static final Logger logger = LogManager.getLogger();
+	
+	private static Connection connect = null;
+	private static Statement statement = null;
+	private PreparedStatement preparedStatement = null;
+	private static ResultSet resultSet = null;
+	
+	public static void execute(String username, String password, String host, String releaseNumber) throws Exception {
+		logger.info("Running MapOldStableIds");
+		// Need to use mysql driver to access stable_identifiers db
+		logger.info("Connecting to stable_identifers db...");
+		Class.forName("com.mysql.jdbc.Driver");
+		connect = DriverManager.getConnection("jdbc:mysql://" + host + "/stable_identifiers?" + "user=" + username + "&password=" + password);
+		statement = connect.createStatement();
+		resultSet = statement.executeQuery("SELECT identifier,instanceId FROM StableIdentifier");
+		
+		// Iterate through returned results of DB IDs and stable IDs 
+		logger.info("Mapping Old Stable IDs to Current Stable IDs...");
+		HashMap<String,ArrayList<String>> dbIdToStableIds = new HashMap<String,ArrayList<String>>();
+		ArrayList<String> dbIds = new ArrayList<String>();
+		while (resultSet.next()) {
+			String stableId = resultSet.getString(1);
+			String dbId = resultSet.getString(2);
+			
+			if (dbIdToStableIds.get(dbId) != null) {
+				dbIdToStableIds.get(dbId).add(stableId);
+			} else {
+				ArrayList<String> stableIds = new ArrayList<String>(Arrays.asList(stableId));
+				dbIdToStableIds.put(dbId, stableIds);
+				dbIds.add(dbId);
+			}
+		}
+		Collections.sort(dbIds);
+		
+		// Iterate through array of stable IDs associated with DB ID, splitting into human and non-human groups
+		ArrayList<ArrayList<Object>> hsaIds = new ArrayList<ArrayList<Object>>();
+		ArrayList<ArrayList<Object>> nonHsaIds = new ArrayList<ArrayList<Object>>();
+		for (String dbId : dbIds) {
+			ArrayList<String> stableIds = dbIdToStableIds.get(dbId);
+			Collections.sort(stableIds);
+
+			boolean properStableIdObject = true;
+			if (stableIds.size() < 2 || !stableIds.get(0).matches("^R-.*")) {
+				properStableIdObject = false;
+			}
+			
+			// After sorting the first stable ID in the array is considered the primary ID.
+			// An Array of Arrays is used here, with each interior array's first element being 
+			// the primaryId string and the second element being an array of the remaining stable IDs.
+			if (properStableIdObject) {
+				String primaryId = stableIds.get(0);
+				stableIds.remove(0);
+				ArrayList<Object> organizedIds = new ArrayList<Object>();
```

Comment:
Looking at the way that `organizedIds` is used, it looks like it can have a `String` added to it, or an `ArrayList<String>` added to it, which is why it takes `Object`.

This is a bit odd, and the code that makes use `organizedIds` could probably be rewritten to be a little cleaner, but that might take a bit of time so if it works, I'd leave it alone for now.

Reply:
For sure, we can look at it after the release.